### PR TITLE
Improve table recreation sys check docs

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

- Make sure the examples are tested
- Document version compatibility
- Add hint to avoiding reindex using partitions

Replaces https://github.com/crate/crate/pull/8941

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/admin/system-information.rst [hunk 1]
```diff
@@ -1574,49 +1574,161 @@ Tables need to be recreated
...
+before you can upgrade to a future major version of CrateDB.
 
-If you try to upgrade to the next major version of CrateDB with tables that
-have not been recreated, CrateDB will refuse to start.
+If you try to upgrade to a later major CrateDB version without reindexing the
+tables, CrateDB will refuse to start.
 
-To recreate a table, you have to create new tables, copy over the data and
-rename or remove the old table.
+CrateDB Table Version Compatibility Scheme
```
Comment (mechanomi) [inline docs/admin/system-information.rst]:
```suggestion
CrateDB table version compatibility scheme
```

File: docs/admin/system-information.rst [hunk 2]
```diff
@@ -1574,49 +1574,161 @@ Tables need to be recreated
...
+      - ❌
+      - ✔️
+      - ✔️
+    * - 5.x
+      - ❌
+      - ❌
+      - ✔️
+
+
+Avoiding Reindex Using Partitioned Tables
```
Comment (mechanomi) [inline docs/admin/system-information.rst]:
```suggestion
Avoiding reindex using partitioned tables
```

File: docs/admin/system-information.rst [hunk 3]
```diff
@@ -1574,49 +1574,161 @@ Tables need to be recreated
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Reindexing tables is an expensive operation which can take a long time. If you
+are storing time series data for a certain retention period and delete old
```
Comment (mechanomi) [inline docs/admin/system-information.rst]: might convey your meaning a little better if you phrase it "and intend to delete old data"

File: docs/admin/system-information.rst [hunk 4]
```diff
@@ -1574,49 +1574,161 @@ Tables need to be recreated
...
+data, it is possible to use the :ref:`partitioned_tables` feature to avoid
+reindex operations.
+
+You will have to partition a table by a column that denotes time. For example,
+if you have a retention period of 9 months, you could partition a table by a
```
Comment (mechanomi) [inline docs/admin/system-information.rst]: for numbers 0-9, it's good style to spell them out. so, "nine", in this instance

note: this doesn't apply to literals

File: docs/admin/system-information.rst [hunk 5]
```diff
@@ -1574,49 +1574,161 @@ Tables need to be recreated
...
+``month`` column. Then every month the system creates a new partition. This new
```
Comment (mechanomi) [inline docs/admin/system-information.rst]: how about:

"Then, every month, the system will create a new partition."

File: docs/admin/system-information.rst [hunk 6]
```diff
@@ -1574,49 +1574,161 @@ Tables need to be recreated
...
+partition is created using the active CrateDB version and is compatible with
+the next major CrateDB version. You will also have to delete the oldest
+partition. By doing this you will have rolled through all partitions after 9
```
Comment (mechanomi) [inline docs/admin/system-information.rst]: it's not clear to me how this fits:

"You will also have to delete the oldest partition."

I think maybe this para could start with an "If" condition, i.e., "If you wish to achieve GOAL, then you must..."

that would make it more clear that you are outlining a procedure. as it stands, the bit before this sentence about deleting a partition reads more like a general description of how partitioning works

Comment (mechanomi) [reply docs/admin/system-information.rst]: also, minor: comma after "By doing this"

Comment (mechanomi) [reply docs/admin/system-information.rst]: also, "nine" :)

File: docs/admin/system-information.rst [hunk 7]
```diff
@@ -1574,49 +1574,161 @@ Tables need to be recreated
...
+months and all partitions will be compatible with the next major CrateDB
+version.
+
+
+How To Reindex
```
Comment (mechanomi) [inline docs/admin/system-information.rst]:
```suggestion
How to reindex
```

File: docs/admin/system-information.rst [hunk 8]
```diff
@@ -1574,49 +1574,161 @@ Tables need to be recreated
...
+    ... WHERE table_schema = 'rx' AND table_name = 'metrics';
+    +--------------------+
+    | version['created'] |
+    +--------------------+
+    | 4.2.0              |
+    +--------------------+
+    SELECT 1 row in set (... sec)
 
-    ALTER CLUSTER SWAP TABLE tmp_your_table TO your_table;
+6. Drop the now obsolete old table::
```
Comment (mechanomi) [inline docs/admin/system-information.rst]:
```suggestion
6. Drop the old table, as it is now obsolete::
```

("now obsolete" is a bit confusing as an adjective phrase as it is used here, especially when combined with the additional adjective, "old". if you do want to keep it this way, it should be styled "now-obsolete" to disambiguate the adjectives)

File: docs/admin/system-information.rst (hunk 1 above)
Comment (mechanomi) [reply docs/admin/system-information.rst]: we switched to sentence case for all headings last year (it is considered better style. see parameter one here: https://practicaltypography.com/headings.html)

File: docs/admin/system-information.rst (hunk 8 above)
Comment (mechanomi) [reply docs/admin/system-information.rst]: some related reading here: https://english.stackexchange.com/a/461189 (fun fact: this is one element of typographic style I'm fascinated with and have done far too much research on than I care to admit)

**Comment by mechanomi** [comment]
edit: added some citations 😇

File: docs/admin/system-information.rst (hunk 6 above)
Comment (mfussenegger) [reply docs/admin/system-information.rst]: @mechanomi Tried to rephrase it, maybe you can have another look

Comment (mechanomi) [reply docs/admin/system-information.rst]: @mfussenegger big improvement! thx!

**Pull request merged**